src/renderer/value-transformer/KotlinValueTransformer.ts

```diff
@@ -55,26 +56,30 @@ export class KotlinValueTransformer implements ValueTransformer {
           throw Error('Type not exists');
       }
 
-      return `Map<${keyType}, ${this.convertValueType(valueType.valueType)}>`;
+      return `Map<${keyType}, ${this.convertValueType(valueType.valueType, uniquePath)}>`;
     }
 
     if (isOptionalType(valueType)) {
-      return `${this.convertValueType(valueType.wrappedType)}?`;
+      return `${this.convertValueType(valueType.wrappedType, uniquePath)}?`;
     }
 
     if (isPredefinedType(valueType)) {
       return this.typeNameMap[valueType.name] ?? valueType.name;
     }
 
+    if (isUnionType(valueType)) {
```

Comment:
Since union types are all converted to enum types in `named-types`, why do we still need to handle union types here? I suppose no changes are needed for the renderer.

Reply:
I just learn that when isTupleType is true, the valueType will be rewritten to interface, let me update it